### PR TITLE
ROX-22101: gogo move pkg timestamp utils to protoutils pkg

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

See title.
This PR is a subset of #10325 

PR stack:
- #10525
   - #10526 
      - #10528
        - #10529
          - #10530
            - #10531
              - #10532
                - #10533
                  - #10534
                     - #10535
                        - #10536
                           - #10537 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/protoutils/time.go [hunk 1]
```diff
@@ -26,3 +28,12 @@ func After(ts1, ts2 *types.Timestamp) bool {
...
 }
+
+// RoundTimestamp rounds up ts to the nearest multiple of d. In case of error, the function returns without rounding up.
+func RoundTimestamp(ts *types.Timestamp, d time.Duration) {
+	t, err := protocompat.ConvertTimestampToTimeOrError(ts)
+	if err != nil {
+		return
+	}
+	*ts = *protoconv.ConvertTimeToTimestamp(t.Round(d))
+}
```
Comment (vikin91) [inline pkg/protoutils/time.go:39]: Wouldn't that be easier to test if we would return the rounded result?

File: sensor/common/networkflow/manager/manager_impl_test.go [hunk 2]
```diff
@@ -879,7 +879,7 @@ func (b *sendNetflowsSuite) TestUpdatesGetBufferedWhenUnread() {
 
 	// four times without reading
 	for i := 0; i < 4; i++ {
-		ts := timestamp.NowMinus(time.Duration(4-i) * time.Hour)
+		ts := protoconv.NowMinus(time.Duration(4-i) * time.Hour)
```
Comment (vikin91) [inline sensor/common/networkflow/manager/manager_impl_test.go:882]: We could reverse the for loop to start at 4 and count down to 1 instead using `4-i`.

File: sensor/common/networkflow/manager/manager_impl_test.go [hunk 3]
```diff
@@ -896,7 +896,7 @@ func (b *sendNetflowsSuite) TestCallsDetectionEvenOnFullBuffer() {
 	b.expectDetections(6)
 
 	for i := 0; i < 6; i++ {
-		ts := timestamp.NowMinus(time.Duration(6-i) * time.Hour)
+		ts := protoconv.NowMinus(time.Duration(6-i) * time.Hour)
```
Comment (vikin91) [inline sensor/common/networkflow/manager/manager_impl_test.go:899]: Same here - reversing the for loop would make it easier to read that part.

**Review by vikin91 — APPROVED** [review]

File: sensor/common/networkflow/manager/manager_impl_test.go [hunk 4]
```diff
@@ -895,8 +895,8 @@ func (b *sendNetflowsSuite) TestCallsDetectionEvenOnFullBuffer() {
 	b.expectLookups(6)
 	b.expectDetections(6)
 
-	for i := 0; i < 6; i++ {
-		ts := timestamp.NowMinus(time.Duration(6-i) * time.Hour)
+	for i := 6; i > 0; i-- {
```
Comment (janisz) [inline sensor/common/networkflow/manager/manager_impl_test.go:898]: Why this is better?

Comment (janisz) [reply sensor/common/networkflow/manager/manager_impl_test.go:898]: OK, we do not substract later

**Review by janisz — APPROVED** [review]

**Pull request merged**